### PR TITLE
refactor(eslint): move to newer flatconfig

### PR DESCRIPTION
**Pull request by Requinard**
yeah see the latest commit for the real message.

bumps eslint stuff and adds new rules

### CODE REVIEW

File: src/testUtils.tsx
Comment (lukashaertel) [inline src/testUtils.tsx:34]: Marked as ` error TS2578: Unused '@ts-expect-error' directive.`

**Review by lukashaertel — CHANGES_REQUESTED** [review]

**Comment by Fenrikur** [comment]
Since you are not adhering to [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/), please squash merge this branch once review is done and changes have been approved. (=
I've altered the PR title to conform with Conventional Commits accordingly.

File: .github/workflows/nodejs.yml
Comment (Fenrikur) [inline .github/workflows/nodejs.yml:1]: Would suggest turning
```yaml
on:
  push:
    branches: [ "*" ]
```

into just
```yaml
on:
  push:
```
so the workflow runs on every push.

File: .github/workflows/nodejs.yml
Comment (Fenrikur) [inline .github/workflows/nodejs.yml:17]: Since we're using `pnpm`, might be worth looking into using the appropriate cache setting? (:
```yaml
cache: 'pnpm'
```

**Review by lukashaertel — APPROVED** [review]

**Pull request merged**
